data-prepper-plugins/rds-source/src/test/java/org/opensearch/dataprepper/plugins/source/rds/export/ExportTaskManagerTest.java

```diff
@@ -0,0 +1,121 @@
+        final String prefix = "prefix";
+        final String kmsKey = UUID.randomUUID().toString();
+        final List<String> exportOnly = List.of();
+
+        exportTaskManager.startExportTask(snapshotArn, iamRoleArn, bucket, prefix, kmsKey, exportOnly);
+
+
+        final ArgumentCaptor<StartExportTaskRequest> exportTaskRequestArgumentCaptor =
+                ArgumentCaptor.forClass(StartExportTaskRequest.class);
+
+        verify(rdsClient).startExportTask(exportTaskRequestArgumentCaptor.capture());
+
+        final StartExportTaskRequest actualRequest = exportTaskRequestArgumentCaptor.getValue();
+        assertRequestParameters(actualRequest, snapshotArn, iamRoleArn, bucket, prefix, kmsKey, exportOnly);
+    }
+
+    @Test
+    void test_start_export_task_without_specifying_tables1() {
+        final String snapshotArn = "arn:aws:rds:us-east-1:123456789012:snapshot:snapshot-0b5ae174";
+        final String iamRoleArn = "arn:aws:iam:us-east-1:123456789012:role:my-role";
+        final String bucket = "bucket";
+        final String prefix = "prefix";
+        final String kmsKey = UUID.randomUUID().toString();
+        final List<String> exportOnly = List.of("my_db.cars", "my_db.houses");
+
+        exportTaskManager.startExportTask(snapshotArn, iamRoleArn, bucket, prefix, kmsKey, exportOnly);
+
+        final ArgumentCaptor<StartExportTaskRequest> exportTaskRequestArgumentCaptor =
+                ArgumentCaptor.forClass(StartExportTaskRequest.class);
+
+        verify(rdsClient).startExportTask(exportTaskRequestArgumentCaptor.capture());
+
+        final StartExportTaskRequest actualRequest = exportTaskRequestArgumentCaptor.getValue();
+        assertRequestParameters(actualRequest, snapshotArn, iamRoleArn, bucket, prefix, kmsKey, exportOnly);
+    }
+
+    @Test
+    void test_check_export_status() {
+        final String exportTaskId = UUID.randomUUID().toString();
+        DescribeExportTasksResponse describeExportTasksResponse = mock(DescribeExportTasksResponse.class);
+        when(describeExportTasksResponse.exportTasks()).thenReturn(List.of(ExportTask.builder().status("COMPLETE").build()));
+        when(rdsClient.describeExportTasks(any(DescribeExportTasksRequest.class))).thenReturn(describeExportTasksResponse);
+
+        exportTaskManager.checkExportStatus(exportTaskId);
+
+        final ArgumentCaptor<DescribeExportTasksRequest> exportTaskRequestArgumentCaptor =
+                ArgumentCaptor.forClass(DescribeExportTasksRequest.class);
+
+        verify(rdsClient).describeExportTasks(exportTaskRequestArgumentCaptor.capture());
+
+        final DescribeExportTasksRequest actualRequest = exportTaskRequestArgumentCaptor.getValue();
+        assertThat(actualRequest.exportTaskIdentifier(), equalTo(exportTaskId));
+    }
+
+    private ExportTaskManager createObjectUnderTest() {
+        return new ExportTaskManager(rdsClient);
+    }
+
+    private void assertRequestParameters(final StartExportTaskRequest request,
```

Comment:
I'm not sure that this common method is going to help maintain this code. It is a long list of parameters that can be easy to lose track of.

I'd just duplicate these assertions personally.

An alternative approach - Make an `@Nested` class. Then set these values as fields in a nested `@BeforeEach`. With this, you can make a simpler method that takes a single parameter. It can then assert against the fields.

Reply:
Thanks for the suggestion! I ended up using a parameterized test to combine the two tests and removed this common method.